### PR TITLE
ddl: basic support for invisible index

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve? 

Part of #9246 

### What is changed and how it works?

1. Store `Invisible` Flag in these statements:
	- `CREATE TABLE t (a INT, INDEX (a) INVISIBLE)` 
	- `CREATE INDEX a on t(a) INVISIBLE`
	- `ALTER TABLE t ADD INDEX a INVISIBLE`

2. Add a new column `Visible` in `SHOW INDEX`:
```
mysql> show index from t;
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
| Table | Non_unique | Key_name | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment | Visible | Expression |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
| t     |          1 | a        |            1 | a           | A         |           0 |     NULL | NULL   | YES  | BTREE      |         |               | NO      | NULL       |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
1 row in set (0.00 sec)
```

3. Add a new column `IS_VISIBLE` in table `information_schema.statistics`:
```
mysql> select table_name, index_name, is_visible from information_schema.statistics where table_name = "t";
+------------+------------+------------+
| table_name | index_name | is_visible |
+------------+------------+------------+
| t          | a          | NO         |
+------------+------------+------------+
1 row in set (0.00 sec)

```

4. Show Invisible Index in `SHOW CREATE TABLE`
``` 
mysql> show create table t;
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                     |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` int(11) DEFAULT NULL,
  KEY `a` (`a`) /*!80000 INVISIBLE */
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

 - Increased code complexity

Related changes

 - None

Release note

 - None

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -852,6 +863,9 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
 			cols = append(cols, colInfo)
 		}
 		fmt.Fprintf(buf, "(%s)", strings.Join(cols, ","))
+		if idxInfo.Invisible {
+			fmt.Fprintf(buf, ` /*!80000 INVISIBLE */`)
```
Comment (AilinKid) [inline executor/show.go]: How does comment `80000` come？any reference?

Comment (Deardrops) [reply executor/show.go]: Sorry, I do not find any reference for the `/*!80000 INVISIBLE */`,
but when test in MySQL 8.0, it does have this behavior.

Comment (AilinKid) [reply executor/show.go]: make sense

**Pull request merged**